frontend/src/components/MonacoJsonEditor.tsx

```diff
@@ -0,0 +1,87 @@
+/*
+  This file is part of Edgehog.
+
+  Copyright 2025 SECO Mind Srl
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+  SPDX-License-Identifier: Apache-2.0
+*/
+
+import { useRef, useState, useCallback } from "react";
+import { Editor } from "@monaco-editor/react";
+
+type MonacoJsonEditorProps = {
+  value: string;
+  language?: string;
+  onChange?: (value: string | undefined) => void;
+  defaultValue?: string;
+  readonly?: boolean;
+  initialLines?: number;
+};
+
+const MonacoJsonEditor = ({
+  value,
+  language = "json",
+  onChange,
+  defaultValue,
+  readonly = false,
+  initialLines = 5,
+}: MonacoJsonEditorProps) => {
+  const editorRef = useRef<any>(null);
+  const lineHeight = 22;
+  const [height, setHeight] = useState(initialLines * lineHeight);
+
+  const updateHeight = useCallback(() => {
+    if (editorRef.current) {
+      const contentHeight = editorRef.current.getContentHeight();
+      const minHeight = initialLines * lineHeight;
+      setHeight(Math.max(contentHeight, minHeight));
+      editorRef.current.layout();
+    }
+  }, [initialLines]);
+
+  const handleEditorDidMount = (editor: any) => {
+    editorRef.current = editor;
+    editor.onDidContentSizeChange(updateHeight);
+    updateHeight();
+  };
```

Comment:
Adding a automatic formatter would be a nice addition, this is just a quick suggestion and can be probably improved.

```suggestion
  const formatJson = useCallback(() => {
    if (editorRef.current && autoFormat && language === "json") {
      try {
        const currentValue = editorRef.current.getValue();
        JSON.parse(currentValue); // Validate JSON
        editorRef.current.getAction("editor.action.formatDocument").run();
      } catch (error) {
        // JSON is invalid, don't format
      }
    }
  }, [autoFormat, language]);

  const handleEditorDidMount = (editor: any) => {
    editorRef.current = editor;
    editor.onDidContentSizeChange(updateHeight);
    editor.onDidChangeModelContent(() => {
      setTimeout(formatJson, 100); // Debounce formatting
    });
    updateHeight();
  };
```